### ISSUE TITLE
Query errors are not being visibly reported

### ISSUE THREAD

**Issue by Firehed** [sequelpro/sequelpro#2799]
When a query run from the editor errors, SP shows "Errors" but the errors do not appear to be visible anywhere. They do not show up in the console, and are not reported to the Notification Center like the application's preference appear to indicate they should be. I'm only able to find the actual error by re-running the query directly with the mysql command-line interface. If there's somewhere in the UI that should be displaying the error, it's either way too hard to find or not working correctly.

SP Build 5195 (43926b0), macOS 10.12.4. I have all checkboxes selected in the "Alerts & Logs" preferences, and SP's alert style in Notifications system preferences is "Banners" with all options selected. MySQL 5.6.x server.

**Comment by dmoagx** [comment]
![bildschirmfoto 2017-05-20 um 00 51 33 984](https://cloud.githubusercontent.com/assets/5793864/26269673/80596e46-3cf6-11e7-8393-7e824ee0ddfd.png)
?

**Comment by Firehed** [comment]
Yeah, that whole section is missing for me. Or, looking a bit closer, it seems to be rendering at about ~2px high and I can't do anything to make it properly visible (there's no click/drag target). This is the same in both states of the "toggle query information pane" button (left of the gear under the query results)

**Comment by dmoagx** [comment]
Hm, you can try to download and reinstall the app. Sometimes the updates break things.

**Comment by Firehed** [comment]
I've just updated to the latest nightly (through the built-in updater; now c1930af) and the issue persists. I am assuming that this sufficiently qualifies as reinstallation, but let me know if you think that's insufficient.

This is what I'm seeing:
<img width="1050" alt="screen shot 2017-05-23 at 3 39 38 pm" src="https://cloud.githubusercontent.com/assets/354842/26379488/545735b4-3fce-11e7-806b-cebb702cfe7c.png">

Note the few-pixels-high stripe above the status bar

**Comment by Firehed** [comment]
Purely at random, I discovered this *may* be related to multi-display setups. It only seems to happen when my system is connected to my external displays. I relaunched SP with no external displays connected and the error window renders correctly.

**Comment by dmoagx** [comment]
Please try creating a new user account on your Mac and launch Sequel Pro from there and check if that makes a difference.

**Comment by russianryebread** [comment]
I have the same issue.  It's a regression, as the stable version works just fine on my multi-monitor setup:

*Stable:*
<img width="321" alt="screen shot 2017-07-31 at 12 01 59 pm" src="https://user-images.githubusercontent.com/747085/28789220-0e3b1db4-75ea-11e7-965e-163728a0a1b4.png">

*Nightly:*
<img width="410" alt="screen shot 2017-07-31 at 12 01 52 pm" src="https://user-images.githubusercontent.com/747085/28789228-181a27ee-75ea-11e7-81ac-071807d7e050.png">

**Comment by abhibeckert** [comment]
I have multiple monitors and cannon reproduce the bug.

Could it be related to the version of MySQL? I'm using 5.5.52. @russianryebread what version are you on?

**Comment by abhibeckert** [comment]
Actually never mind, it looks like the error pane is being displayed, but with a height of 7 pixels.

Perhaps this is caused by a forced resize while from a small laptop monitor to a large external one?

**Comment by dmoagx** [comment]
My guess would be that the wrong size has been saved in prefs somewhere.

**Comment by abhibeckert** [comment]
The error pane is not resizable, so shouldn't be saved/restored.

**Comment by russianryebread** [comment]
There is also some strange behavior when you drag the results panel all the way to the bottom (collapsed) and then click on the error pane toggle button:

<img width="25" alt="screen shot 2017-07-31 at 5 01 03 pm" src="https://user-images.githubusercontent.com/747085/28800345-ed8b5d36-7611-11e7-8ccb-72e0a113aac4.png">

Sometimes (intermittently) the 'run current' button even disappears:
<img width="525" alt="screen shot 2017-07-31 at 5 02 04 pm" src="https://user-images.githubusercontent.com/747085/28800399-1be9043a-7612-11e7-919c-0149d28a261e.png">

Interestingly enough, while testing, it resolved itself using the same server, same computer, and same connections (no error display in one tab, and when I opened another tab it started working in the new tab.)  Then I restarted it, and it is working in all tabs, so there is definitely something intermittent happening.

It's definitely related to pulling the results panel down and toggling the error pane, though.

Sorry I don't have a better replication path... :(

**Comment by abhibeckert** [comment]
> There is also some strange behavior when you drag the results panel all the way to the bottom (collapsed) and then click on the error pane toggle button

I am able to reproduce this.

**Comment by abhibeckert** [comment]
I'm adding this to the 1.2 release for now, but since 1.2 is scheduled for release candidate this week we might push it back to 1.2.1.

**Comment by dopesong** [comment]
This one is really annoying.. :(

**Comment by gboudreau** [comment]
This issue is related to the new tabs. I did a git bisect, and found this commit to be when this regression was introduced:
```
92ee758b8c2d75782ed4b83ac56068606ef5b27a is the first bad commit
commit 92ee758b8c2d75782ed4b83ac56068606ef5b27a
Author: Abhi Beckert <abhi@abhibeckert.com>
Date:   Fri Mar 18 16:48:10 2016 +1000

    #346 Work on new tab bar design
```

I'll try to dig further, see what I can understand from what was changed, and how it affected the error pane. If you have some ideas/pointers, they would be most welcome.

**Comment by gboudreau** [comment]
PR sent; I couldn't find a better way than checking if there was a draggable handle associated with the splitView, to know if the separator was draggable or not. Maybe there is a better way..?

**Comment by dmoagx** [comment]
@gboudreau 
I think the real question is why `[self _lengthOfView:[[self subviews] objectAtIndex:collapsibleSubviewIndex]]` is not zero as it should be when first opening a connection.

From doing a quick look it seems there is an error somewhere in `-adjustSubviews` that makes the view become larger than 0 when it should not and this seems to be related to the tab bar animation.

**Comment by dmoagx** [comment]
After looking further I think #3087 is the cause here.
During animation there is a moment where the content area height is e.g. `577.5pt` (which is a valid size on a retina screen) resulting in the query editor view and the info pane getting sizes of `577.5pt + 0pt`. But as explained in #3087 the code will try to round this in a non-dpi-aware fashion which ends up as `577pt + 0.5pt`.

**Comment by 80S66GN5** [comment]
For others who run in to this before it is fixed, I've found the easiest way to fix it is to restart Sequel Pro, force an error before selecting a database, and then select your database.

**Comment by gboudreau** [comment]
Quick note: While my PR might not be the right way to fix this issue (preventing one of the side-effects, instead of resolving the root cause), I've been using it since June without noticeable side effects. i.e. it seems like a useful workaround, until the root cause of this issue gets fixed.  
I'm still building my own binary because I need that fix.

**Comment by dmoagx** [comment]
this should be fixed in 98cec9a

**Comment by gboudreau** [comment]
Yes, that seems to have fixed the error panel not showing bug. Thanks.
